Azure/sonic-buildimage
# Multi-ASIC implementation

**- What I did**
To support SONiC on multi-asic platforms, the approach is to create SONiC docker stack for each ASIC. Each SONiC docker stack is separated by network namespaces for each ASIC.
Each network namespace will comprise of key dockers : 
database
swss
syncd
bgp
teamd
interfaces-config
lldp

**- How I did it**
Main changes done to support SONiC on multi-asic platform are:
1. systemd service template for multi-instance service is added to start N number of systemd services where N is the number of ASICs in the platform.
2. Changes in FRR config to support BGP between the ASIC namespaces. To support this, a new type is added in DEVICE_METADATA called "InternalFrontend" and "InternalBackend" to differentiate between a front-end ASIC and back-end ASIC.
3. Change in docker_img_ctl.j2 to support starting dockers with a specific ASIC instance number which will be passed to the <docker_name>.sh script by the systemd service.
4. Apart from multiple instances of database docker, a global database docker will exist in the host namespace.

**- How to verify it**
Build a multi-asic VS image.
Follow instruction (in README.vsd) to bring up multi-asic VS image by adding the number of ASICs in  device/platform/asic.conf and also adding a specific topology.sh in HWSKU directory to create a virtual ASIC connectivity in the VS.
Once the VS image comes up, ensure the dockers come up and verify BGP sessions between ASICs.

**- Description for the changelog**



**- A picture of a cute animal (not mandatory but encouraged)**
